File: CLLIX.TAAuditTracker.Domain/AppUser.cs
using Microsoft.AspNetCore.Identity;

namespace CLLIX.TAAuditTracker.Domain
{
    public class AppUser : IdentityUser
    {
        // Optional: Add domain-specific fields
        public string FullName { get; set; } = string.Empty;
        public string Department { get; set; } = string.Empty;
        public string Role { get; set; } = string.Empty;
        public bool IsActive { get; set; }
        public DateTime CreatedDate { get; set; } = DateTime.UtcNow;
        public DateTime? ModifiedDate { get; set; } = DateTime.UtcNow;
        public required string CreatedBy { get; set; }

        // Navigation properties (if needed)
        public ICollection<BookingReservation> Reservations { get; set; } = new List<BookingReservation>();
    }
}

File: CLLIX.TAAuditTracker.ClientUi/src/app/_interfaces/app-user.ts
import { BookingReservation } from './booking-reservation';

export interface AppUser {
  id: string; // from IdentityUser
  userName: string;
  email: string;
  phoneNumber?: string;
  emailConfirmed?: boolean;
  phoneNumberConfirmed?: boolean;
  twoFactorEnabled?: boolean;
  lockoutEnabled?: boolean;
  lockoutEnd?: string;
  accessFailedCount?: number;

  fullName: string;
  department: string;
  role: string;
  isActive: boolean;
  createdDate: string;
  modifiedDate?: string;
  createdBy: string;

  reservations: BookingReservation[];
}
